### ISSUE TITLE
Added columns not getting restored.

### ISSUE THREAD

**Issue by settysreekanth** [angular-ui/ng-grid#3476]
Newly added columns are not getting restored. Follow the below steps to reproduce the issue,

1. Add new column (Click Add)
2. save grid state in localstorage
3. Change state of grid but don't save state
4. Now restore grid state by clicking Restore button and this fetches settings from localstorage (working)
5. However refresh the browser.
6. Restore grid state from localstorage (Not working)

I have created plunk for this issue http://plnkr.co/edit/mR7lbG?p=preview

**Comment by PaulL1** [comment]
I think the problem is that you're making an assumption that restore state will add columns that aren't there.  Restore will only add settings to columns it can find - the base column needs to be there. 

If you change the plunker such that the columns are there but hidden, then it works.  http://plnkr.co/edit/c0Xwl8QWgS6RWr7aQUsX?p=preview

**Comment by settysreekanth** [comment]
@PaulL1  Thanks for your reply. I modified my plunker but I have to click restore button twice to restore my actual settings. 

http://plnkr.co/edit/g6hGjz?p=preview
Steps:
1. Click Add twice to add 2 columns like Company and Company2
2. Sort Asc or Desc for Company column and Click save state button.
3.Refresh the page and click restore state button twice to restore state (including sorting on company column). 

Here clicking restore state first time will restore columns and 2nd time will restore actual settings. 

Could you please help to find out the issue here?

**Comment by settysreekanth** [comment]
@PaulL1 could you please suggest ?

**Comment by c0bra** [comment]
@PaulL1 seems to me that restore state should restore everything at once if it's able to, would you agree?

**Comment by PaulL1** [comment]
Yes, it should, but I suspect it's more a timing/usage issue.  I'll take a look later today.

**Comment by HarnikNoa** [comment]
Was this issue ever resolved?  I'm running into a similar issue, where it takes me two button clicks ("reset grid") to restore to the correct grid setting.  The first restores (or clears, in this case) the filters, the second moves the columns back.
